fineract-core/src/main/java/org/apache/fineract/commands/service/SynchronousCommandProcessingService.java

```diff
@@ -103,32 +102,30 @@ public CommandProcessingResult executeCommand(final CommandWrapper wrapper, fina
         }
         exceptionWhenTheRequestAlreadyProcessed(wrapper, idempotencyKey, isRetry);
 
-        boolean sameTransaction = BatchRequestContextHolder.getEnclosingTransaction().isPresent();
         if (commandSource == null) {
             AppUser user = context.authenticatedUser(wrapper);
-            commandSource = sameTransaction ? commandSourceService.saveInitialSameTransaction(wrapper, command, user, idempotencyKey)
-                    : commandSourceService.saveInitialNewTransaction(wrapper, command, user, idempotencyKey);
-            storeCommandIdInContext(commandSource); // Store command id as a request attribute
+            commandSource = commandSourceService.getInitialCommandSource(wrapper, command, user, idempotencyKey);
```

Comment:
This looks not correct. If we are not storing in the DB a command processing was started, there is absolutely no defense to start processing the same command multiple times. Am I missing something?

Reply:
action_name, entity_name, idempotency_key is a compound unique key, cannot be saved twice. That is true that the idempotency_key is not unique, as I thought, it is.